### PR TITLE
[SPARK-12600][SQL] Remove deprecated methods in Spark SQL

### PR DESCRIPTION
**Pull request by rxin**

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/LegacyTypeStringParser.scala [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+import scala.util.parsing.combinator.RegexParsers
+
+import org.apache.spark.sql.types._
+
+/**
+ * Parser that turns case class strings into datatypes. This is only here to maintain compatibility
+ * with Parquet files written by Spark 1.1 and below.
+ */
+object LegacyTypeStringParser extends RegexParsers {
```
Comment (rxin) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/LegacyTypeStringParser.scala]: This is moved from DataType.scala. It is kept here so Spark 2.0 can still read parquet files written by Spark 1.1.

File: python/pyspark/sql/readwriter.py [hunk 2]
```diff
@@ -179,7 +177,17 @@ def json(self, path, schema=None):
         elif type(path) == list:
             return self._df(self._jreader.json(self._sqlContext._sc._jvm.PythonUtils.toSeq(path)))
         elif isinstance(path, RDD):
-            return self._df(self._jreader.json(path._jrdd))
+            def func(iterator):
```
Comment (rxin) [inline python/pyspark/sql/readwriter.py]: cc @davies this fixed a bug before with read.json.

**Comment by marmbrus** [comment]
LGTM

File: sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala [hunk 3]
```diff
@@ -126,6 +115,16 @@ class DataFrameReader private[sql](sqlContext: SQLContext) extends Logging {
   }
 
   /**
+   * Loads input in as a [[DataFrame]], for data sources that require a path (e.g. data backed by
+   * a local or distributed file system).
+   *
+   * @since 1.4.0
+   */
+  def load(path: String): DataFrame = {
```
Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala]: this one is actually useful for both R / Python, and required for data sources that don't support paths. So I'm just removing the todo to remove it.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/types/DecimalType.scala [hunk 4]
```diff
@@ -20,25 +20,10 @@ package org.apache.spark.sql.types
...
 import org.apache.spark.annotation.DeveloperApi
-import org.apache.spark.sql.AnalysisException
 import org.apache.spark.sql.catalyst.ScalaReflectionLock
 import org.apache.spark.sql.catalyst.expressions.Expression
 
 
-/** Precision parameters for a Decimal */
-@deprecated("Use DecimalType(precision, scale) directly", "1.5")
-case class PrecisionInfo(precision: Int, scale: Int) {
-  if (scale > precision) {
```
Comment (davies) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/types/DecimalType.scala]: We should keep this checking somewhere

**Comment by davies** [comment]
LGTM except one comment.

**Comment by marmbrus** [comment]
To avoid conflicts I'm going to go ahead and merge this.  @rxin can address comments in a followup.

**Pull request closed without merging**